.github/workflows/wipac-cicd.yml

```diff
@@ -36,44 +43,67 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - id: versions
-        uses: WIPACrepo/wipac-dev-py-versions-action@v1
+        uses: WIPACrepo/wipac-dev-py-versions-action@v2
 
   tests:
     needs: [py-versions]
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
-      matrix: ${{ fromJSON(needs.py-versions.outputs.matrix) }}
+      matrix:
+        version: ${{ fromJSON(needs.py-versions.outputs.matrix) }}
+        broker: ["", pulsar, rabbitmq, gcp, nats]
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v3
         with:
-          python-version: ${{ matrix.py3_versions }}
-      - name: Pytest (Default)
-        run: |
-          pip install --upgrade pip wheel setuptools pytest
-          pip install .
-          # pytest -vvvv
-      - name: Pytest [Pulsar]
-        run: |
-          pip install --upgrade pip wheel setuptools pytest
-          pip install .[pulsar]
-          # pytest -vvvv
-      - name: Pytest [RabbitMQ]
-        run: |
-          pip install --upgrade pip wheel setuptools pytest
-          pip install .[rabbitmq]
-          # pytest -vvvv
-      - name: Pytest [GCP]
-        run: |
-          pip install --upgrade pip wheel setuptools pytest
-          pip install .[gcp]
-          # pytest -vvvv
-      - name: Pytest [NATS]
+          python-version: ${{ matrix.version }}
+      - name: Integrate
         run: |
+          echo "--------------------------------------------------------------"
+          echo "starting broker..."
+
+          # Pulsar
+          if [ -z "${{ matrix.broker }}" ] || [ "${{ matrix.broker }}" = "pulsar" ]; then
+            wget https://raw.githubusercontent.com/WIPACrepo/MQClient/master/resources/docker-pulsar.sh
+            chmod +x ./docker-pulsar.sh
```

Comment:
Instead of doing a wget, should this be a second repo checkout?  You can checkout into a subdir.

Reply:
Oh, that's pretty cool. I didn't know that was possible